src/axom/quest/examples/quest_step_file.cpp

```diff
@@ -0,0 +1,2223 @@
+// Copyright (c) 2017-2024, Lawrence Livermore National Security, LLC and
+// other Axom Project Developers. See the top-level LICENSE file for details.
+//
+// SPDX-License-Identifier: (BSD-3-Clause)
+
+#include "axom/config.hpp"
+#include "axom/core.hpp"
+#include "axom/slic.hpp"
+#include "axom/primal.hpp"
+#include "axom/mint.hpp"
+
+#include "axom/CLI11.hpp"
+#include "axom/fmt.hpp"
+
+#include "opencascade/BRepAdaptor_Curve.hxx"
+#include "opencascade/BRepBuilderAPI_MakeFace.hxx"
+#include "opencascade/BRepBuilderAPI_NurbsConvert.hxx"
+#include "opencascade/BRepMesh_IncrementalMesh.hxx"
+#include "opencascade/BRepTools.hxx"
+#include "opencascade/BRep_Tool.hxx"
+#include "opencascade/Geom2d_BSplineCurve.hxx"
+#include "opencascade/Geom2d_Curve.hxx"
+#include "opencascade/Geom2dConvert.hxx"
+#include "opencascade/GeomAbs_CurveType.hxx"
+#include "opencascade/Geom_BSplineSurface.hxx"
+#include "opencascade/Geom_RectangularTrimmedSurface.hxx"
+#include "opencascade/Geom_Surface.hxx"
+#include "opencascade/Interface_Static.hxx"
+#include "opencascade/Poly_Triangulation.hxx"
+#include "opencascade/Precision.hxx"
+#include "opencascade/STEPControl_Reader.hxx"
+#include "opencascade/TColgp_Array1OfPnt.hxx"
+#include "opencascade/TColgp_Array2OfPnt.hxx"
+#include "opencascade/TopAbs.hxx"
+#include "opencascade/TopExp.hxx"
+#include "opencascade/TopExp_Explorer.hxx"
+#include "opencascade/TopLoc_Location.hxx"
+#include "opencascade/TopTools_IndexedMapOfShape.hxx"
+#include "opencascade/TopoDS.hxx"
+#include "opencascade/TopoDS_Edge.hxx"
+#include "opencascade/TopoDS_Face.hxx"
+#include "opencascade/TopoDS_Shape.hxx"
+#include "opencascade/TopoDS_Wire.hxx"
+#include <iostream>
+
+/**
+ * /file quest_step_file.cpp
+ * /brief Example that loads in a STEP file and converts the surface patches and curves to Axom's NURBS representations
+ *
+ * This example reads in STEP files representing trimmed NURBS meshes using OpenCASCADE, 
+ * converts the patches and trimming curves to Axom's NURBSPatch and NURBSCurve primitives, 
+ * and generates various outputs including SVG and STL files.
+ *
+ * /note This example requires Axom to be configured with OpenCASCADE enabled.
+ */
+
+/// Struct to hold data associated with each surface patch of the mesh
+struct PatchData
+{
+  int patchIndex {-1};
+  bool wasOriginallyPeriodic_u {false};
+  bool wasOriginallyPeriodic_v {false};
+  axom::primal::NURBSPatch<double, 3> nurbsPatch;
+  axom::primal::BoundingBox<double, 2> parametricBBox;
+  axom::primal::BoundingBox<double, 3> physicalBBox;
+  axom::Array<axom::primal::NURBSCurve<double, 2>> trimmingCurves;
+  axom::Array<bool> trimmingCurves_originallyPeriodic;
+};
+
+/**
+ * Class to read in a STEP file representing trimmed NURBS meshes using OpenCASCADE 
+ * and convert the patches and trimming curves to axom's NURBSPatch and NURBSCurve primitives.
+ * 
+ * Implementation note: Since axom's primitives do not support periodic knots, 
```

Comment:
```suggestion
 * and convert the patches and trimming curves to Axom's NURBSPatch and NURBSCurve primitives.
 * 
 * Implementation note: Since Axom's primitives do not support periodic knots, 
```